pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+}
+
+func resourceLdapSettingsDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	rClient := m.(*resty.Client)
+	ldapConfigs := &xmlLdapConfig{}
+
+	rsrcLdapSetting := unpackLdapSetting(d)
+
+	response, err := rClient.R().Get("artifactory/api/system/configuration")
+	if err != nil {
+		return diag.Errorf("failed to retrieve data from <base_url>/artifactory/api/system/configuration during Read")
+	}
+
+	err = xml.Unmarshal(response.Body(), &ldapConfigs)
+	if err != nil {
+		return diag.Errorf("failed to xml unmarshal ldap setting during delete operation")
+	}
+
+	var restoreLdapSettings = map[string]map[string]map[string]LdapSetting{}
+	restoreLdapSettings["security"] = map[string]map[string]LdapSetting{}
+	restoreLdapSettings["security"]["ldapSettings"] = map[string]LdapSetting{}
+
+	for _, iterLdapSetting := range ldapConfigs.LdapSettings.LdapSettingArr {
+		if iterLdapSetting.Key != rsrcLdapSetting.Key {
+			restoreLdapSettings["security"]["ldapSettings"][iterLdapSetting.Key] = iterLdapSetting
+		}
+	}
+
+	var clearAllLdapSettingsConfigs = `
+security:
+  ldapSettings: ~
+`
+	err = sendConfigurationPatch([]byte(clearAllLdapSettingsConfigs), m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during Delete for clearing all Ldap Settings")
+	}
+
+	restoreRestOfLdapSettingsConfigs, err := yaml.Marshal(&restoreLdapSettings)
+	if err != nil {
+		return diag.Errorf("failed to marshal ldap settings during Update")
+	}
+
+	err = sendConfigurationPatch([]byte(restoreRestOfLdapSettingsConfigs), m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during restoration of Ldap Settings")
+	}
+	return nil
+}
+
+func unpackLdapSetting(s *schema.ResourceData) *LdapSetting {
+	d := &ResourceData{s}
+	ldapSetting := new(LdapSetting)
+	ldapSetting.Key = d.getString("key", false)
+	ldapSetting.Enabled = d.getBool("enabled", false)
+	ldapSetting.LdapUrl = d.getString("ldap_url", false)
+	ldapSetting.AutoCreateUser = d.getBool("auto_create_user", false)
+	ldapSetting.LdapPoisoningProtection = d.getBool("ldap_poisoning_protection", false)
+	ldapSetting.PagingSupportEnabled = d.getBool("paging_support_enabled", false)
+	ldapSetting.AllowUserToAccessProfile = d.getBool("allow_user_to_access_profile", false)
+	ldapSetting.UserDnPattern = d.getString("user_dn_pattern", false)
+	ldapSetting.EmailAttribute = d.getString("email_attribute", false)
+	ldapSetting.Search.SearchSubTree = d.getBool("search_sub_tree", false)
+	ldapSetting.Search.SearchBase = d.getString("search_base", false)
+	ldapSetting.Search.SearchFilter = d.getString("search_filter", false)
+	ldapSetting.Search.ManagerDn = d.getString("manager_dn", false)
+	ldapSetting.Search.ManagerPassword = d.getString("manager_password", true)
+	return ldapSetting
+}
+
+func packLdapSetting(ldapSetting *LdapSetting, d *schema.ResourceData) diag.Diagnostics {
+	setValue := mkLens(d)
+	setValue("key", ldapSetting.Key)
```

Comment:
You know about the `universalPack` function?

Reply:
Moved to universalPack function.